### DISCUSSION TITLE
web implementation

### DISCUSSION THREAD

**Discussion by bkbartk** [hassio-addons/addon-vscode#402] [Ideas]
If I understand this correctly
"Visual Studio Code runs as a remote server using code-server, and is a fully fledged VSCode experience."
Visual Studio code is running server side and then provisioned to the web using some kind of meganism.

I know VS code has a fully featured web equivalent.
https://code.visualstudio.com/docs/editor/vscode-web
Is it somehow possible to implement this web version.
I know this probably means a major change in code, but if it is possible it would have a couple of benefits like server resources and architecture limitations.

**Comment by frenck** [comment]
Those are different things, I think your question is about the Home Assistant Configuration Helper vscode extension actually? As in, this add-on runs code-server 🤷

**Comment by bkbartk** [comment]
the question actually is if it is possible to run the "Studio Code Server" add on client side so all calculations are done by the browser instead of the server.
the add-on is really heavy on ram usage.
![image](https://user-images.githubusercontent.com/17070805/159517558-692d6f44-1835-4213-95c8-d9b3ee20eeed.png)
which makes sense while running server side.
But if the add on could run client side it only needs to provide html/js/css and maybe some streaming data. and it really could save some memory.
of course the browser will consume more.
and I don't know if this is possible at all.

**Reply by frenck** [reply]
Your question don't make any sense to me, sorry. These things are not related. If you run VSCode in the browser, you don't need this add-on at all? Feel free to do so, not something that is related to this add-on at all. 🤷  Please note, the add-on runs code-server, which uses the upstream codebase from vscode (including their web interface).

**Comment by bkbartk** [comment]
but if I run it in browser I still need to be able to access files on the home assistant system.
![image](https://user-images.githubusercontent.com/17070805/159521804-81264b45-61bc-43b0-91c2-f8d6d6884297.png)
without it being an add on this won't be possible I think.

But I think I misunderstand something completely and I don't want to waste your time on that part. So this one can be closed.

**Comment by frenck** [comment]
> but if I run it in browser I still need to be able to access files on the home assistant system.

That is not a concern for this add-on, nor a goal, not can it be a goal
